### ISSUE TITLE
 fancy_pypi_readme: TypeError: build_text() missing 1 required positional argument: 'version'

### ISSUE THREAD

**Issue by njzjz** [scikit-build/scikit-build-core#592]
I got this error just now:

```
    Running command Preparing metadata (pyproject.toml)
    Traceback (most recent call last):
      File "/opt/python/cp311-cp311/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
        main()
      File "/opt/python/cp311-cp311/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/python/cp311-cp311/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 152, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/tmp/pip-build-env-5hc5unxx/overlay/lib/python3.11/site-packages/scikit_build_core/build/__init__.py", line 72, in prepare_metadata_for_build_wheel
        return _build_wheel_impl(
               ^^^^^^^^^^^^^^^^^^
      File "/tmp/pip-build-env-5hc5unxx/overlay/lib/python3.11/site-packages/scikit_build_core/build/wheel.py", line 134, in _build_wheel_impl
        metadata = get_standard_metadata(pyproject, settings)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/tmp/pip-build-env-5hc5unxx/overlay/lib/python3.11/site-packages/scikit_build_core/settings/metadata.py", line 36, in get_standard_metadata
        new_pyproject_dict["project"][field] = provider.dynamic_metadata(field, config)
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/tmp/pip-build-env-5hc5unxx/overlay/lib/python3.11/site-packages/scikit_build_core/metadata/fancy_pypi_readme.py", line 39, in dynamic_metadata
        "text": build_text(config.fragments, config.substitutions)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    TypeError: build_text() missing 1 required positional argument: 'version'
    error: subprocess-exited-with-error
    
    × Preparing metadata (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> See above for output.
```

https://github.com/scikit-build/scikit-build-core/blob/2f080f8079d2b39e6166fd59dcd0025a1dcfea80/src/scikit_build_core/metadata/fancy_pypi_readme.py#L39

It seems the upstream API was changed:
- https://github.com/hynek/hatch-fancy-pypi-readme/pull/39

**Comment by hynek** [comment]
this should be fixed by hatch-fancy-pypi-readme 24.1.0 that is now on PyPI

**Comment by njzjz** [comment]
I confirm it works. Thanks!
